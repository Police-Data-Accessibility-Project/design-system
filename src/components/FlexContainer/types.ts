export interface PdapFlexContainerProps {
	alignment?: 'center' | 'start';
	className?: string;
	component?: string;
}
