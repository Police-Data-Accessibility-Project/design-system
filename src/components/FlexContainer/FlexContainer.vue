<template>
	<component :is="component" :class="classes">
		<slot />
	</component>
</template>

<script setup lang="ts">
// Imports
import { reactive } from 'vue';

// Types
import { PdapFlexContainerProps } from './types';

// Props
const props = withDefaults(defineProps<PdapFlexContainerProps>(), {
	alignment: 'start',
	component: 'div',
});

// CSS class map
const classes = reactive({
	'pdap-flex-container': true,
	[`pdap-flex-container-${props.alignment}`]: true,
	[String(props.className)]: Boolean(props.className),
});
</script>

<style scoped>
@tailwind components;

@layer components {
	/* Layout */
	.pdap-flex-container {
		/* Left and top aligned by default */
		@apply items-start flex h-full justify-start flex-col max-w-none mx-auto p-8 w-full;

		/* Responsiveness */
		@apply xs:max-w-3xl lg:max-w-4xl;
	}

	.pdap-flex-container-center {
		/* Center-aligned (vert & hor) */
		@apply pdap-flex-container items-center justify-center;
	}
	.pdap-flex-container-start {
		@apply pdap-flex-container;
	}

	.pdap-flex-container > * {
		@apply w-full;
	}
}
</style>
