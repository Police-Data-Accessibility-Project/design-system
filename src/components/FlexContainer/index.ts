import FlexContainer from './FlexContainer.vue';

export { FlexContainer };
