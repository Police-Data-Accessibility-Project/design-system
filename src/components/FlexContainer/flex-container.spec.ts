// Component
import FlexContainer from './FlexContainer.vue';

// Utils
import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';

// Test
describe('Renders container component', () => {
	// Render
	test('Renders a container', () => {
		const wrapper = mount(FlexContainer, {
			slots: {
				// Todo: Render some actual components here to test
				default: 'Container Content',
			},
		});

		expect(wrapper.find('.pdap-flex-container').exists()).toBe(true);
		expect(wrapper.classes()).toContain('pdap-flex-container');
		expect(wrapper.html()).toContain('Container Content');
		expect(wrapper.html()).toMatchSnapshot();
	});

	// Props
	// Props - align
	test('Renders start aligned container', () => {
		const wrapper = mount(FlexContainer, { props: { alignment: 'start' } });
		expect(wrapper.props().alignment).toBe('start');
		expect(wrapper.classes()).toContain('pdap-flex-container-start');
	});

	test('Renders center aligned container', () => {
		const wrapper = mount(FlexContainer, { props: { alignment: 'center' } });
		expect(wrapper.props().alignment).toBe('center');
		expect(wrapper.classes()).toContain('pdap-flex-container-center');
	});
});
