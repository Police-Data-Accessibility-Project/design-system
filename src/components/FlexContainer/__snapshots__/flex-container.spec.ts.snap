// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders container component > Renders a container 1`] = `<div class="pdap-container pdap-container-start">Container Content</div>`;
